etc/benchmarks/main.mjs

```diff
@@ -98,6 +98,37 @@ await runner({
   }
 });
 
+/// Batch full of user doc with 20 char strings w/ 20 strings
+/// nextBatch simulate
+/// nextBatch: [ { string * 20 } * 1000 ] /// Garbage call
+await runner({
+  skip: false,
+  name: 'deserialize a large batch of documents each with an array of many strings',
+  iterations,
+  setup(libs) {
+    const bson = libs[0].lib;
+    return bson.serialize({
+      nextBatch: Array.from({ length: 1000 }, () => ({
+        _id: new bson.ObjectId(),
+        arrayField: Array.from({ length: 20 }, (_, i) => '5e99f3f5d3ab06936d36000' + i)
+      }))
+    });
+  },
+  async run(i, bson, document) {
+    await Promise.all(
+      Array.from(
+        { length: 100 },
+        (_, i) =>
+          new Promise(resolve => {
+            setTimeout(() => {
+              resolve(bson.lib.deserialize(document, { validation: { utf8: false } }));
+            }, 20);
+          })
+      )
```

Comment:
What is the purpose of the setTimeout here?  And why did you choose a setTimeout of 20ms specifically?

Reply:
It manufactures the delay that the driver does when we promise.all many many finds, 20ms is arbitrary, kept it low so the bench runs in a reasonable amount of time